File: C++11/백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <stdio.h>

int gcd(int a,int b){ //최대공약수 구하는 함수(유클리드 호제법)
    if(b==0)return a;
    return gcd(b,a%b);
}
int main(){
    int a,b,d,m;
    scanf("%d %d",&a,&b);
    if(b>a){
        int temp = a;
        a = b;
        b = temp;
    }
    d = gcd(a,b);
    m = a*b/d; //최소공배수 = (두 수의 곱) / 최소공약수
    printf("%d\n%d",d,m);
}

File: Java/백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.Scanner;

public class Main{
    public static int gcd(int a,int b){
        if(b==0){
            return a;
        }
        return gcd(b,a%b);
    }
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        int b = in.nextInt();
        if(b>a){
            int temp = a;
            a = b;
            b = temp;
        }
        int d = gcd(a,b);
        int m = a*b/d;
        System.out.println(d);
        System.out.println(m);
    }
}